### ISSUE TITLE
Possible to globally exclude patterns or literal strings somehow?

### ISSUE THREAD

**Issue by luckman212** [koalaman/shellcheck#2859]
I am wondering if there's any way to exclude specific patterns (fixed strings, globs or regexes) from triggering [SC2154](https://github.com/koalaman/shellcheck/wiki/SC2154)

#### Situation

I export certain variables for formatting ANSI bold, colors etc in my `.bash_profile`
But, most of my scripts trigger SC2154 because shellcheck doesn't know about these declared vars. I don't want to completely disable the check, because it's very useful! But, I just want to exclude a few specific vars or strings from triggering it.

E.g. in `.bash_profile`:
```bash
export Fg=$(printf '\x01\e[1;32m\x02')   # green
export Fn=$(printf '\x01\e[0m\x02')      # normal
```

And dumb example script
```bash
#!/bin/bash
echo "Hello ${Fg}green${Fn}"  # <== triggers SC2154 warning
```

In that case I would love if there was a `~/.shellcheckignore` file or something where I could put literal strings

```
${Fg}
${Fn}
```

Thank you 🙏

**Comment by luckman212** [comment]
Just adding, I noticed and am aware that all-uppercase vars do not trigger this warning as long as `enable=check-unassigned-uppercase` is not set in `.shellcheckrc`. But I have hundreds of scripts with the mixed-case vars, so it would be a real chore to replace them all. If that's the only option then so be it.

**Comment by brother** [comment]
> But I have hundreds of scripts with the mixed-case vars, so it would be a real chore to replace them all. If that's the only option then so be it.

It would indeed be easier for **you** to have someone else change shellcheck instead of you changing your scripts :stuck_out_tongue_winking_eye:

**Comment by luckman212** [comment]
Yes @brother I see your point. And of course, that is true. I wasn't trying to be snarky with this question, I was merely asking if it was possible before setting out on that somewhat rough road.
